### PR TITLE
GH-790 Fix duration format

### PR DESCRIPTION
**Pull request by vLuckyyy**
Fixes: #790. This is currently the only solution that satisfies me.

The idea is that the format should return a single member of the format "1s" but also take into account the milli-seconds at a later time, so that the format does not return empty text.

image

### CODE REVIEW

File: eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java [hunk 1]
```diff
@@ -7,18 +7,39 @@
 
 public class DurationUtil {
 
-    private static final TemporalAmountParser<Duration> WITHOUT_MILLS = new DurationParser()
+    private static final TemporalAmountParser<Duration> WITHOUT_MILLS_FORMAT = new DurationParser()
```
Comment (Jakubk15) [inline eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java:10]:
```suggestion
    private static final TemporalAmountParser<Duration> WITHOUT_MILLIS_FORMAT = new DurationParser()
```

File: eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java [hunk 2]
```diff
@@ -7,18 +7,39 @@
...
+
+    public static final Duration ONE_SECOND = Duration.ofSeconds(1);
+
     public DurationUtil() {
         throw new UnsupportedOperationException("This class cannot be instantiated");
     }
 
+    public static String format(Duration duration, boolean removeMills) {
+        if (removeMills) {
+            return WITHOUT_MILLS_FORMAT.format(duration);
```
Comment (Jakubk15) [inline eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java:32]:
```suggestion
            return WITHOUT_MILLIS_FORMAT.format(duration);
```

File: eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java [hunk 3]
```diff
@@ -7,18 +7,39 @@
...
+        .withUnit("s", ChronoUnit.SECONDS)
+        .withUnit("ms", ChronoUnit.MILLIS);
+
+    public static final Duration ONE_SECOND = Duration.ofSeconds(1);
+
     public DurationUtil() {
         throw new UnsupportedOperationException("This class cannot be instantiated");
     }
 
+    public static String format(Duration duration, boolean removeMills) {
```
Comment (Jakubk15) [inline eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java:30]:
```suggestion
    public static String format(Duration duration, boolean removeMillis) {
```

File: eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java [hunk 4]
```diff
@@ -7,18 +7,39 @@
...
+        if (removeMills) {
```
Comment (Jakubk15) [inline eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java:31]:
```suggestion
        if (removeMillis) {
```

File: eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java [hunk 5]
```diff
@@ -7,18 +7,39 @@
...
+            return WITHOUT_MILLS_FORMAT.format(duration);
+        }
+
+        if (duration.toMillis() < ONE_SECOND.toMillis()) {
+            return STANDARD_FORMAT.format(duration);
+        }
+
+        return WITHOUT_MILLS_FORMAT.format(duration);
```
Comment (Jakubk15) [inline eternalcore-core/src/main/java/com/eternalcode/core/util/DurationUtil.java:39]:
```suggestion
        return WITHOUT_MILLIS_FORMAT.format(duration);
```

**Review by Jakubk15 — APPROVED** [review]

**Review by Rollczi — COMMENTED** [review]
Naprawić commons

**Pull request merged**